array-manipulation-1/src/compact.js

```diff
@@ -0,0 +1,10 @@
+/* exported compact */
+
+function compact(array) {
+  for (var i = 0, result = []; i < array.length; i++) {
+    if (array[i] !== false && array[i] !== 0 && array[i] !== null &&
+      array[i] !== undefined && array[i] !== '' &&
+      !Number.isNaN(array[i])) result.push(array[i]);
```

Comment:
This works! But we can simplify it by noticing that the values we want to exclude from the array are all false values. Which means we can just check if the current array value is not false. 

if(array[i]){
result.push(array[i])
}

great initial solution!